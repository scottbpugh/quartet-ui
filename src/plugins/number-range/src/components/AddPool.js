// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import {connect} from "react-redux";
import {
  Panels,
  RightPanel,
  LeftPanel
} from "../../../../components/layouts/Panels";
import {Card} from "@blueprintjs/core";
import NRTree from "./NRTree";
import PoolForm from "./PoolForm";
import {FormattedMessage} from "react-intl";

class _AddPool extends Component {
  constructor(props) {
    super(props);
    this.currentServer = this.props.nr[this.props.match.params.serverID];
  }
  componentDidMount() {}
  render() {
    return (
      <Panels title={<FormattedMessage id="plugins.numberRange.addPool" />}>
        <RightPanel>
          <div className="large-cards-container">
            <Card className="pt-elevation-4 form-card">
              <h5>
                <FormattedMessage id="plugins.numberRange.addPool" />
              </h5>
              <PoolForm
                server={this.currentServer.server}
                history={this.props.history}
              />
            </Card>
          </div>
        </RightPanel>
      </Panels>
    );
  }
}

export const AddPool = connect((state, ownProps) => {
  return {
    servers: state.serversettings.servers,
    nr: state.numberrange.servers
  };
}, {})(_AddPool);
