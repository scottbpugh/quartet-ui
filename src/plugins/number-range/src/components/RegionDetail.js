// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import {connect} from "react-redux";
import {loadRegions} from "../reducers/numberrange";
import {Panels, RightPanel, LeftPanel} from "components/layouts/Panels";
import {Card, Callout} from "@blueprintjs/core";
import RegionRange from "./RegionRange";
import {setAllocation} from "../reducers/numberrange";
import "../style.css";
import classNames from "classnames";
import {FormattedDate, FormattedMessage, FormattedNumber} from "react-intl";

/**
 * _RegionDetail - Description
 * @extends Component
 */
class _RegionDetail extends Component {
  /*static propTypes = {
    history: PropTypes.object.isRequired
  };*/
  constructor(props) {
    super(props);
    this.state = {alloc: 1, lastUpdated: null};
    // these two properties below make it easy to retrieve
    // and trigger actions.
    this.currentPool = {readable_name: ""};
    this.currentServer = {};
  }
  componentDidMount() {
    this.loadPoolDetail(this.props);
  }
  previewAlloc = evt => {
    this.setState({alloc: Number(evt.target.value)});
  };
  loadPoolDetail(props) {
    let nrServer = props.nr[props.match.params.serverID];
    this.currentServer = nrServer.server;
    for (let pool of nrServer.pools) {
      if (pool.machine_name === props.match.params.pool) {
        this.currentPool = pool;
      }
    }
    this.props.loadRegions(this.currentServer, this.currentPool);
  }
  setAllocation = evt => {
    this.props.setAllocation(
      this.currentServer,
      this.currentPool,
      this.state.alloc
    );
  };
  componentWillReceiveProps(nextProps) {
    if (nextProps.match.params.pool !== this.props.match.params.pool) {
      this.loadPoolDetail(nextProps);
      return;
    }
    // shake card for the last updated region.
    /*nextProps.currentRegions.map((region, index) => {
      if (
        this.props.currentRegions[index] &&
        region.state !== this.props.currentRegions[index].state
      ) {
        console.log(
          "new state",
          region.state,
          "old state",
          this.props.currentRegions[index].state
        );
        this.setState({lastUpdated: region.machine_name}, () => {
          window.setTimeout(() => {
            this.setState({lastUpdated: null});
          }, 3000);
        });
      }
      return null;
    });*/
  }

  render() {
    let regions = this.props.currentRegions;

    return (
      <RightPanel
        title={
          <FormattedMessage
            id="plugins.numberRange.regionDetailTitle"
            values={{poolName: this.currentPool.readable_name}}
          />
        }>
        <div className="auto-cards-container">
          {regions && regions.length > 0 ? (
            regions.map(region => (
              <Card
                className={classNames({
                  "pt-elevation-4": true,
                  "region-detail": true,
                  updated: this.state.lastUpdated === region.machine_name
                })}
                key={region.machine_name}>
                <h5>{region.readable_name}</h5>
                <ul>
                  <li>
                    <FormattedMessage id="plugins.numberRange.createdOn" />:{" "}
                    <FormattedDate value={region.created_date} />
                  </li>
                  <li>
                    <FormattedMessage id="plugins.numberRange.status" />:{" "}
                    {region.active ? (
                      <FormattedMessage id="plugins.numberRange.active" />
                    ) : (
                      <FormattedMessage id="plugins.numberRange.inactive" />
                    )}
                  </li>
                  <li>
                    <FormattedMessage
                      id="plugins.numberRange.range"
                      defaultMessage="Range"
                    />: <FormattedNumber value={region.start} />{" "}
                    <FormattedMessage id="plugins.numberRange.to" />{" "}
                    <FormattedNumber value={region.end} />
                  </li>
                  <li>
                    <FormattedMessage id="plugins.numberRange.state" />:{" "}
                    <FormattedNumber value={region.state} />
                  </li>
                </ul>
                <RegionRange
                  start={region.start}
                  end={region.end}
                  state={region.state}
                  alloc={this.state.alloc}
                />
              </Card>
            ))
          ) : (
            <Callout>
              <FormattedMessage id="plugins.numberRange.noRegionInPool" />
            </Callout>
          )}
        </div>
      </RightPanel>
    );
    //<div>{JSON.stringify(this.props.region)}</div>;
  }
}

export var RegionDetail = connect(
  (state, ownProps) => {
    return {
      servers: state.serversettings.servers,
      currentRegions: state.numberrange.currentRegions,
      nr: state.numberrange.servers
    };
  },
  {loadRegions, setAllocation}
)(_RegionDetail);
