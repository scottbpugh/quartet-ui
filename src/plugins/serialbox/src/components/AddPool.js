// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import PoolForm from "./PoolForm";
import {deleteResponseRule} from "../reducers/numberrange";
const React = qu4rtet.require("react");
const {Component} = React;
const {connect} = qu4rtet.require("react-redux");
const {RightPanel} = qu4rtet.require("./components/layouts/Panels");
const {Card, Button, ButtonGroup} = qu4rtet.require("@blueprintjs/core");
const {FormattedMessage} = qu4rtet.require("react-intl");

class _AddPool extends Component {
  constructor(props) {
    super(props);
    this.currentServer = this.props.nr[this.props.match.params.serverID];
  }
  componentDidMount() {}

  editResponseRule = responseRule => {
    let pool = this.getPool();
    this.props.history.push({
      pathname: `/number-range/add-response-rule/${
        this.currentServer.server.serverID
      }/pool-id/${pool.id}`,
      state: {defaultValues: responseRule, edit: true, pool: pool}
    });
  };
  deleteResponseRule = responseRule => {
    let pool = this.getPool();
    this.props.deleteResponseRule(this.currentServer.server, responseRule);
  };
  getEditMode = () => {
    return (this.props.location &&
      this.props.location.state &&
      this.props.location.state.editPool) ||
      this.props.match.params.poolName
      ? true
      : false;
  };
  getPool = () => {
    let pool = {};
    let editMode = this.getEditMode();
    if (editMode) {
      try {
        pool = this.props.location.state.defaultValues;
      } catch (e) {
        // location state may not be available.
      }
    }
    if (this.props.match.params.poolName) {
      editMode = true;
      let pools = this.props.nr[this.currentServer.server.serverID].pools;
      // most up to date.
      pool = pools.find(pool => {
        return pool.machine_name === this.props.match.params.poolName;
      });
    }
    return pool;
  };
  render() {
    let editMode = this.getEditMode();
    let pool = this.getPool();
    return (
      <RightPanel
        title={
          !editMode ? (
            <FormattedMessage id="plugins.numberRange.addPool" />
          ) : (
            <FormattedMessage id="plugins.numberRange.editPool" />
          )
        }>
        <div className="large-cards-container">
          <Card className="pt-elevation-4 form-card">
            <h5>
              {!editMode ? (
                <FormattedMessage id="plugins.numberRange.addPool" />
              ) : (
                <FormattedMessage id="plugins.numberRange.editPool" />
              )}
            </h5>
            <PoolForm
              server={this.currentServer.server}
              history={this.props.history}
              pool={pool}
            />
          </Card>
          {editMode ? (
            <Card className="pt-elevation-4 form-card">
              <h5>
                <button
                  className="pt-button right-aligned-elem pt-interactive pt-intent-primary"
                  onClick={e => {
                    this.props.history.push({
                      pathname: `/number-range/add-response-rule/${
                        this.currentServer.server.serverID
                      }/pool-id/${pool.id}/`,
                      state: {pool: pool}
                    });
                  }}>
                  <FormattedMessage id="plugins.numberRange.addResponseRule" />
                </button>
                <FormattedMessage id="plugins.numberRange.responseRules" />
              </h5>
              {pool &&
              Array.isArray(pool.response_rules) &&
              pool.response_rules.length > 0 ? (
                <table className="pt-table pt-interactive pt-bordered pt-striped">
                  <thead>
                    <tr>
                      <th>
                        <FormattedMessage
                          id="plugins.numberRange.ruleName"
                          defaultMessage="Rule Name"
                        />
                      </th>
                      <th>
                        {" "}
                        <FormattedMessage
                          id="plugins.numberRange.contentType"
                          defaultMessage="Content Type"
                        />
                      </th>
                      <th />
                    </tr>
                  </thead>
                  <tbody>
                    {pool.response_rules
                      ? pool.response_rules.map(responseRule => {
                          return (
                            <tr key={responseRule.id}>
                              <td>
                                {responseRule.rule
                                  ? this.props.rules.find(
                                      rule => rule.id === responseRule.rule
                                    ).name
                                  : null}
                              </td>
                              <td>{responseRule.content_type}</td>
                              <td style={{width: "80px"}}>
                                <ButtonGroup minimal small>
                                  <Button
                                    small="true"
                                    iconName="edit"
                                    onClick={this.editResponseRule.bind(
                                      this,
                                      responseRule
                                    )}
                                  />
                                  <Button
                                    small="true"
                                    iconName="trash"
                                    onClick={this.deleteResponseRule.bind(
                                      this,
                                      responseRule
                                    )}
                                  />
                                </ButtonGroup>
                              </td>
                            </tr>
                          );
                        })
                      : null}
                  </tbody>
                </table>
              ) : null}
            </Card>
          ) : null}
        </div>
      </RightPanel>
    );
  }
}

export const AddPool = connect(
  (state, ownProps) => {
    return {
      servers: state.serversettings.servers,
      nr: state.numberrange.servers,
      rules: state.capture.servers
        ? state.capture.servers[ownProps.match.params.serverID].rules
        : []
    };
  },
  {deleteResponseRule}
)(_AddPool);
