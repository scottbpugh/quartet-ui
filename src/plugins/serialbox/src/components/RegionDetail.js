// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {RegionCard} from "./RegionCard";
import "../style.css";
const React = qu4rtet.require("react");
const {Component} = React;
const {connect} = qu4rtet.require("react-redux");
const {pluginRegistry} = qu4rtet.require("./plugins/pluginRegistration");
const {withRouter} = qu4rtet.require("react-router");
const {Callout} = qu4rtet.require("@blueprintjs/core");
const {FormattedMessage} = qu4rtet.require("react-intl");
const classNames = qu4rtet.require("classnames");
const {DeleteDialog} = qu4rtet.require("./components/elements/DeleteDialog");
const {RightPanel} = qu4rtet.require("./components/layouts/Panels");
import {loadPools, loadRegions} from "../reducers/numberrange";

/**
 * _RegionDetail - Description
 * @extends Component
 */
class _RegionDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {alloc: 1, lastUpdated: null};
    // these two properties below make it easy to retrieve
    // and trigger actions.
    this.currentPool = {readable_name: "", machine_name: ""};
  }
  componentDidMount() {
    //this.props.loadPools(this.props.servers[this.props.match.params.serverID]);
    this.loadPoolDetail(this.props);
  }
  previewAlloc = evt => {
    this.setState({alloc: Number(evt.target.value)});
  };
  loadPoolDetail(props) {
    for (let pool of this.props.pools) {
      if (pool.machine_name === props.match.params.pool) {
        this.currentPool = pool;
      }
    }
    this.props.loadRegions(
      pluginRegistry.getServer(props.server.serverID),
      this.currentPool
    );
  }

  render() {
    let regions = this.props.currentRegions;

    return (
      <RightPanel
        title={
          <FormattedMessage
            id="plugins.numberRange.regionDetailTitle"
            values={{poolName: this.currentPool.readable_name}}
          />
        }>
        <div className="auto-cards-container">
          {regions && regions.length > 0 ? (
            regions.map(region => (
              <RegionCard
                key={region.machine_name}
                lastUpdated={this.state.lastUpdated}
                region={region}
                alloc={this.state.alloc}
                pool={this.currentPool}
                serverID={this.props.server.serverID}
                serverObject={pluginRegistry.getServer(
                  this.props.server.serverID
                )}
                history={this.props.history}
              />
            ))
          ) : (
            <Callout>
              <div className="pt-running-text">
                <FormattedMessage id="plugins.numberRange.noRegionInPool" />
              </div>
            </Callout>
          )}
        </div>
      </RightPanel>
    );
    //<div>{JSON.stringify(this.props.region)}</div>;
  }
}

export var RegionDetail = connect(
  (state, ownProps) => {
    return {
      server: state.serversettings.servers[ownProps.match.params.serverID],
      pools: state.numberrange.servers[ownProps.match.params.serverID].pools,
      currentRegions: state.numberrange.currentRegions
    };
  },
  {loadPools, loadRegions}
)(_RegionDetail);
