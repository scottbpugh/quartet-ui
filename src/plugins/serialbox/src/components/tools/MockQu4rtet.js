// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
// Fake local storage.
const qu4rtetRequire = require("qu4rtet-ui/lib/require.js").default;

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value;
  }

  removeItem(key) {
    delete this.store[key];
  }
}
window = global;
export const localStorage = new LocalStorageMock();

// add fake local storage for tests.
window.localStorage = localStorage;

// add fake require for ipcRenderer.
window.require = module => {
  if (module === "path") {
    return {
      join: () => {
        return "";
      }
    };
  } else if (module === "electron") {
    return {
      ipcRenderer: {
        on: () => {},
        send: () => {}
      },
      remote: {
        require: path => {
          if (path === "./main-process/plugin-manager.js") {
            return {
              install: () => {
                return {done: "ok"};
              }
            };
          }
          return {};
        },
        app: {
          getVersion: () => {
            return "test-version";
          },
          getPath: () => {
            return "somepath";
          }
        }
      }
    };
  } else {
    try {
      return qu4rtetRequire(module);      
    } catch(e) {
      return null;
    }

  }
};

const {qu4rtet} = require("qu4rtet-ui/lib/qu4rtet");
const React = qu4rtetRequire("react");
const {Component} = React;
