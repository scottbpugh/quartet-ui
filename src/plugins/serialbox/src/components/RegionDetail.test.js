// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import "tools/MockQu4rtet.js";
const {
  TestWrapper,
  mockStore,
  initialState,
  updateRegistryIntl
} = window.require("./tools/mockStore");
const React = window.require("react");
const renderer = require("react-test-renderer");
const Router = window.require("react-router-dom").MemoryRouter;
const {Provider} = window.require("react-redux");
import {AddPool} from "./AddPool";
const {Server} = window.require("./lib/servers");
const {pluginRegistry} = window.require("./plugins/pluginRegistration");
const {flattenMessages} = window.require("./lib/flattenMessages");
const messages = window.require("./messages").default;
import nrmessages from "../messages";
import {RegionDetail} from "./RegionDetail";

let locale = "en-US";
const newIntl = {
  ...initialState.intl,
  messages: flattenMessages({...messages[locale], ...nrmessages[locale]})
};

const pluginData = {
  ...initialState,
  serversettings: {servers: {fakeid: {serverID: "fakeid"}}},
  numberrange: {
    servers: {
      fakeid: {
        server: {
          serverID: "fakeid",
          password: "toor",
          username: "root",
          port: "80",
          serverName: "localhost",
          serverSettingName: "fake server",
          ssl: false,
          path: ""
        },
        pools: [
          {
            sequentialregion_set: [],
            created_date: "2018-02-12T14:56:11.462232Z",
            modified_date: "2018-02-12T14:56:11.462279Z",
            readable_name: "Fake Pool",
            machine_name: "fakepool",
            active: true,
            request_threshold: 50000
          }
        ]
      }
    },
    region: {},
    currentRegions: []
  },
  intl: newIntl
};

const store = mockStore(pluginData);

it("renders correctly a pool with no region", () => {
  let server = pluginData.numberrange.servers.fakeid.server;
  pluginRegistry.registerServer(new Server(server));
  const promise = Promise.resolve({
    statusCode: 200,
    ok: true,
    body: []
  });
  window.fetch = jest.fn().mockImplementation(() => promise);
  const regionDetailScreen = renderer
    .create(
      <TestWrapper locale={locale} messages={newIntl.messages} store={store}>
        <RegionDetail
          store={store}
          match={{params: {serverID: "fakeid", pool: "fakepool"}}}
          server={server}
        />
      </TestWrapper>
    )
    .toJSON();
  return promise.then(data => {
    console.log("Triggered", data);
    expect(regionDetailScreen).toMatchSnapshot();
  });
});
