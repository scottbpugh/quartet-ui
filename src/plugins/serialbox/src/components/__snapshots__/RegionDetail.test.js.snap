// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a pool with no region 1`] = `
<div>
  <div
    className="right-panel"
  >
    <div>
      <div
        className="auto-cards-container"
      >
        <div
          className="pt-callout"
        >
          <div
            className="pt-running-text"
          >
            <span>
              There is currently no region in this pool.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
